SemanticUI-less blocks widget (ObjectListWidget) + Improve and reorder documentation using diataxis. (#486)
kitconcept/volto-light-theme: docs/conceptual-guides/requirements-and-specifications.md

> # Requirements and specifications

Reviewer: Done!

# Requirements and specifications

Volto Light Theme sets the following requirements and specifications for its development.

## Must not use any Semantic UI component or styling

Semantic UI as a design component system is deprecated in Volto 18 and will be removed in Plone 7.

> In preparation for this, VLT uses no Semantic UI component, nor any related styling (`.ui.XXX`).

Reviewer 2: ```suggestion
In preparation for this, VLT uses no Semantic UI component, nor any related styling (`.ui.XXX` CSS classes).
```

The Plone 7 strategy includes the following principles.

-   Provide a very basic, structural, vanilla collection of components to build upon theming and the CMS UI, as implemented in [`@plone/components`](https://github.com/plone/volto/tree/main/packages/components).

> -   These components will be based upon a headless component system [React Aria Components](https://react-spectrum.adobe.com/react-aria/components.html).

Reviewer: 🥳

-   Volto projects can be themed using `@plone/components` as a base, or use a completely different design or component system of the developer's or integrator's choice.

## Volto components shadowing (`customizations`) use case and best practice

> If possible, we will switch to Semantic UI-less components when `@plone/components` is ready, especially if the elements that we customize are clearly theme-related, such as the header, footer, and other theme elements.

Reviewer: When I see "less" in this context, I think "LESS", which ain't what you mean. It can also mean less Semantic UI, which might be might you mean. I'm not sure.

```suggestion
If possible, we will switch to Semantic UI-free components when `@plone/components` is ready, especially if the customized elements are clearly theme-related, such as the header, footer, and other theme elements.
```

When other Volto customizations are not clearly part of the theme, such as the Search block, it's fine to stick with what the original uses, even Semantic UI.
When Volto makes the switch in the future, we should adapt all the customizations to match the one in the Volto core.

```{important}

> VLT takes the approach of using a proxy to a component of the `components` folder.

Reviewer: ```suggestion
VLT takes the approach of using a proxy to a component of the {file}`components` folder.
```

This way it's easier to keep track of changes, and another add-on can customize further the VLT component, not the original Volto customization.
```

## Layout based in kitconcept's FZJ/DLR projects

Starting with FZJ/DLR projects, we've been trying a new concept in layout for Volto.
This layout uses three widths for the content elements:

-   Narrow (text)
-   Default (blocks)
-   Layout (main screen elements such as Header, Footer)

> The Layout sized elements snap to 1440px.

Reviewer: ```suggestion
The Layout sized elements snaps to 1440px.
```

> The breakpoints are also different from default Volto.

Reviewer: ```suggestion
The breakpoints are also different from Volto's defaults.
```

This layout uses mixins and CSS that can be found in {file}`layout.less` in the {file}`src/theme` folder.
It leverages container queries for managing the layout styling.

## Organization of the files

> We will start organizing the files in the root of the {file}`theme` folder, to differentiate them from a normal "Semantic UI" theme.

Reviewer: ```suggestion
We will start organizing the files in the root of the {file}`theme` folder, to differentiate them from a normal Semantic UI theme.
```

> Take a look at the current state.
> We will follow this convention:

Reviewer: ```suggestion
Use the following organization conventions.
```
